tests/test_multi_file_viewer.ts

```diff
@@ -0,0 +1,999 @@
+        add_first_file_button.trigger('click');
+
+        let view_file_component = multi_file_viewer.find({ref: 'view_file_component'});
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(0);
+        expect(multi_file_viewer.vm.$data.file_names_and_content.size).toEqual(1);
+        expect(multi_file_viewer.vm.$data.files_currently_viewing.length).toEqual(1);
+        expect(view_file_component.text()).toContain(wrapper.vm.$data.file_1_content);
+
+        let add_second_file_button = wrapper.find(".second-button");
+        add_second_file_button.trigger('click');
+        view_file_component = multi_file_viewer.find({ref: 'view_file_component'});
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(1);
+        expect(multi_file_viewer.vm.$data.file_names_and_content.size).toEqual(2);
+        expect(multi_file_viewer.vm.$data.files_currently_viewing.length).toEqual(2);
+        expect(view_file_component.text()).toContain(wrapper.vm.$data.file_2_content);
+
+        let add_third_file_button = wrapper.find(".third-button");
+        add_third_file_button.trigger('click');
+        view_file_component = multi_file_viewer.find({ref: 'view_file_component'});
+
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(2);
+        expect(multi_file_viewer.vm.$data.file_names_and_content.size).toEqual(3);
+        expect(multi_file_viewer.vm.$data.files_currently_viewing.length).toEqual(3);
+        expect(view_file_component.text()).toContain(wrapper.vm.$data.file_3_content);
+
+        let add_fourth_file_button = wrapper.find(".fourth-button");
+        add_fourth_file_button.trigger('click');
+        view_file_component = multi_file_viewer.find({ref: 'view_file_component'});
+        expect(multi_file_viewer.vm.$data.file_names_and_content.size).toEqual(4);
+        expect(multi_file_viewer.vm.$data.files_currently_viewing.length).toEqual(4);
+        expect(multi_file_viewer.vm.$data.active_tab_index).toEqual(3);
+        expect(view_file_component.text()).toContain(wrapper.vm.$data.file_4_content);
+    });
+
+    test('Clicking on a tab makes it the active tab', () => {
+        @Component({
+            template:  `<div>
+                          <p class="first-button" @click="open_file(1)"> Add Kiwi </p>
+                          <p class="second-button" @click="open_file(2)"> Add Lime </p>
+                          <p class="third-button" @click="open_file(3)"> Add Mango </p>
+                          <p class="fourth-button" @click="open_file(4)"> Add Nectarine </p>
+                          <div class="mvf-container">
+                            <multi_file_viewer
+                              ref="multi_file">
+                            </multi_file_viewer>
+                          </div>
+                        </div>`,
+            components: {
+                'multi_file_viewer': MultiFileViewer
+            }
+        })
+
+        class WrapperComponent extends Vue {
```

Comment:
If this component is the same in multiple tests, you can move it up to the test file level rather than duplicating it